test/softwallet.js

```diff
@@ -0,0 +1,56 @@
+/* globals describe,it,before,after */
+const assert = require('assert');
+const web3 = require('web3');
+const web3c = require('../');
+const gateway = require('./mockgateway');
+const artifact = require('../demo/example.json');
+const web3cmocksigner = require('./web3c').web3cMockSigner;
+
+describe('Web3', () => {
+
+  let gw;
+  let timeout;
+
+  if (process && process.env && process.env.TIMEOUT) {
+    timeout = process.env.TIMEOUT;
+  } else {
+    timeout = 5000;
+  }
+
+  let provider = gateway.start();
+
+  before(function(done) {
+    provider.listen(0, () => {
+      gw = new web3.providers.HttpProvider('http://localhost:' + provider.address().port);
+      done();
+    });
+  });
+  after(() => {
+    provider.close();
+  });
+
+  it('should deploy and transact with a soft wallet', async () => {
+    let _web3c = web3cmocksigner(gw);
+    let acct = _web3c.eth.accounts.create();
+    _web3c.confidential.defaultAccount = acct.address;
+    _web3c.confidential.accounts.wallet.add(acct);
+
+    let contract = new _web3c.confidential.Contract(artifact.abi);
+    contract = await contract.deploy({
+      data: artifact.bytecode
+    }).send({
+      from: acct.address,
+      gas: '0x3b9aca00',
+      gasPrice: '0x3b9aca00'
+    });
+    if (!contract) {
+      throw new Error('contract should not be empty');
+    }
+    let mthd = contract.methods.incrementCounter();
+    await mthd.send({
+      gas: '0x3b9aca00',
```

Comment:
Do we have to specify the gas here for any reason? In the past it was symptomatic of a bug.

Reply:
the accounts signer will make a request out to the mock gateway to estimate gas if this isn't specified. this wrapping should cause that to happen after encryption, so it'll get the correct gas for the confidential operation. This was easier that plumbing through the extra method in the mock gateway.

Reply:
👍 sounds reasonable.